File: backend/doc_configs/mutual_nda.py
from typing import Optional, Literal
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Mutual Non-Disclosure Agreement (MNDA).

## Fields you must collect (all are required)

1. **Purpose** -- how confidential information may be used
   Ask: "What is the purpose of sharing confidential information? (e.g. 'Evaluating a potential acquisition', 'Exploring a technology partnership')"

2. **Effective Date** -- agreement start date
   Ask: "What is the effective date? (e.g. 'today', 'March 15 2026', 'next Monday' -- I'll convert it to the right format)"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15", "15/03/2026"), convert to YYYY-MM-DD format directly. Accept any reasonable date format the user provides.

3. **MNDA Term** -- how long the agreement lasts
   Ask: "How long should this NDA last? (e.g. 'expires after 2 years', 'continues until either party terminates it')"
   Set type to "expires" with a year count, or "continues" if open-ended.
   Shorthands and mappings:
   - "1yr", "2yr", "3yrs", "1y", "2y", "1 year", "2 years", "3 years", "one year", "two years", "three years" -> expires, that many years
   - "18 months", "24 months" etc. -> expires, round to nearest whole year (18 months = 2 years, 24 months = 2 years)
   - "open", "open-ended", "until terminated", "rolling", "no expiry", "never expires", "indefinitely" -> continues
   - For any other duration, use your best judgment to map to expires with a year count or continues

4. **Term of Confidentiality** -- how long shared information stays confidential
   Ask: "How long should the confidential information remain protected after the NDA ends? (e.g. '3 years', 'forever')"
   Set type to "years" with a count, or "perpetuity".
   Shorthands and mappings:
   - "1yr", "2yr", "3yrs", "1y", "2y", "1 year", "2 years", "3 years", "one year", "two years", "three years" -> years, that many
   - "18 months", "24 months" etc. -> years, round to nearest whole year
   - "forever", "perpetuity", "perp", "always", "indefinitely", "never expires", "no expiry" -> perpetuity
   - For any other duration, use your best judgment to map to years with a count or perpetuity

5. **Governing Law** -- the state or jurisdiction whose laws govern the agreement
   Ask: "Which state or jurisdiction's law should govern this agreement?"
   Accept any jurisdiction the user provides, including non-US states, territories, or countries -- capture it exactly as given.

6. **Jurisdiction** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes?"
   Accept any court location the user provides, including non-US cities, states, or countries -- capture it exactly as given.

7. **Party 1** -- name, title, company, notice address (email or postal), signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of Acme Corp, jane@acme.com, signing today"

8. **Party 2** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum -- do not dump all questions at once
- Always include a concrete example in parentheses when asking a question, so the user knows the expected format
- When the user answers, confirm what you captured then always end your message with a complete, fully-formed question -- never trail off with phrases like "Now we need...", "Next up...", or "Let's move on to..." without finishing the sentence
- If you need more information from the user, always ask a clear follow-up question
- Fields considered "empty" and must be asked: purpose (if empty string), effectiveDate (if empty), mndaTerm (if years is 0), termOfConfidentiality (if years is 0), governingLaw (if empty), jurisdiction (if empty), party1/party2 sub-fields (if empty strings)
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "start over", "clear the date", "clear party 1"):
- To clear a **specific field**, return that field set to its empty value in "fields" (e.g. `{{"governingLaw": ""}}` or `{{"party1": {{"name": "", "title": "", "company": "", "noticeAddress": "", "date": ""}}}}`) and confirm what was cleared, then ask the user to re-provide it
- To clear **all fields**, return every field reset to empty values and say you have cleared the form, then ask the first question again
- Never clear silently -- always confirm what was cleared in "message"

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialMndaTerm(BaseModel):
    type: Optional[Literal["expires", "continues"]] = None
    years: Optional[int] = None


class PartialTermOfConfidentiality(BaseModel):
    type: Optional[Literal["years", "perpetuity"]] = None
    years: Optional[int] = None


class PartialFields(BaseModel):
    purpose: Optional[str] = None
    effectiveDate: Optional[str] = None
    mndaTerm: Optional[PartialMndaTerm] = None
    termOfConfidentiality: Optional[PartialTermOfConfidentiality] = None
    governingLaw: Optional[str] = None
    jurisdiction: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields

File: frontend/lib/schemas/mutual-nda.ts
import { DocSchema } from "../doc-schema";

export const mutualNdaSchema: DocSchema = {
  id: "mutual-nda",
  title: "Mutual Non-Disclosure Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Mutual Non-Disclosure Agreement (the "MNDA") consists of: (1) this Cover Page and (2) the Common Paper Mutual NDA Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/mutual-nda/1.0. Any modifications of the Standard Terms should be made on the Cover Page, which will control over conflicts with the Standard Terms.',
  coverFields: [
    {
      key: "purpose",
      label: "Purpose",
      hint: "How Confidential Information may be used",
      type: "text",
      placeholder: "Enter purpose...",
    },
    {
      key: "effectiveDate",
      label: "Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "mndaTerm",
      label: "MNDA Term",
      hint: "The length of this MNDA",
      type: "checkbox-group",
      options: [
        { value: "expires", label: "Expires {years} from Effective Date", hasYears: true },
        { value: "continues", label: "Continues until terminated in accordance with the terms of the MNDA" },
      ],
    },
    {
      key: "termOfConfidentiality",
      label: "Term of Confidentiality",
      hint: "How long Confidential Information is protected",
      type: "checkbox-group",
      options: [
        {
          value: "years",
          label: "{years} from Effective Date, but in the case of trade secrets until Confidential Information is no longer considered a trade secret under applicable laws",
          hasYears: true,
        },
        { value: "perpetuity", label: "In perpetuity" },
      ],
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "jurisdiction",
      label: "Jurisdiction",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Party 1", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Party 2", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this MNDA as of the Effective Date.",
  templateFile: "templates/Mutual-NDA.md",
  footerNote: "Common Paper Mutual Non-Disclosure Agreement (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your NDA assistant. I'll help you fill in the Mutual Non-Disclosure Agreement through conversation. Let's start -- what's the purpose of sharing confidential information between the two parties? (e.g. 'Evaluating a potential acquisition', 'Exploring a technology partnership', 'Discussing joint-development of a new product')",
  pdfFilename: "mutual-nda.pdf",
};
